File: lib/backend/src/routes/logs.ts
import { Elysia } from 'elysia';
import { LoggerService } from '../services/LoggerService';

export const logsRoutes = new Elysia({ prefix: '/logs' })
  .get('/', () => {
    const logs = LoggerService.getLogs();
    return {
      success: true,
      logs,
      count: logs.length
    };
  })
  .get('/recent', ({ query }: { query: { limit?: string } }) => {
    const limit = query.limit ? parseInt(query.limit) : 50;
    const logs = LoggerService.getLogs(limit);
    return {
      success: true,
      logs,
      count: logs.length
    };
  })
  .get('/since', ({ query }) => {
    const timestamp = query.timestamp as string | undefined;
    if (!timestamp) {
      return {
        success: false,
        error: 'timestamp query parameter required'
      };
    }

    const logs = LoggerService.getLogsSince(timestamp);
    return {
      success: true,
      logs,
      count: logs.length
    };
  })
  .get('/level/:level', ({ params }: { params: { level: string } }) => {
    const validLevels = ['info', 'warn', 'error', 'debug'];
    if (!validLevels.includes(params.level)) {
      return {
        success: false,
        error: `Invalid level. Must be one of: ${validLevels.join(', ')}`
      };
    }

    const logs = LoggerService.getLogsByLevel(params.level as 'info' | 'warn' | 'error' | 'debug');
    return {
      success: true,
      logs,
      count: logs.length
    };
  })
  .delete('/', () => {
    LoggerService.clearLogs();
    return {
      success: true,
      message: 'All logs cleared'
    };
  });

File: lib/backend/src/routes/logs.py
from fastapi import APIRouter
from ..services.LoggerService import LoggerService

router = APIRouter(prefix="/logs")

@router.get("/")
def get_logs():
    logs = LoggerService.get_logs()
    return {
        "success": True,
        "logs": logs,
        "count": len(logs)
    }

@router.get("/recent")
def get_recent_logs(limit: int = 50):
    logs = LoggerService.get_logs(limit)
    return {
        "success": True,
        "logs": logs,
        "count": len(logs)
    }

@router.get("/since")
def get_logs_since(timestamp: str = None):
    if not timestamp:
        return {
            "success": False,
            "error": "timestamp query parameter required"
        }
    logs = LoggerService.get_logs_since(timestamp)
    return {
        "success": True,
        "logs": logs,
        "count": len(logs)
    }

@router.get("/level/{level}")
def get_logs_by_level(level: str):
    valid_levels = ["info", "warn", "error", "debug"]
    if level not in valid_levels:
        return {
            "success": False,
            "error": f"Invalid level. Must be one of: {', '.join(valid_levels)}"
        }
    logs = LoggerService.get_logs_by_level(level)
    return {
        "success": True,
        "logs": logs,
        "count": len(logs)
    }

@router.delete("/")
def clear_logs():
    LoggerService.clear_logs()
    return {
        "success": True,
        "message": "All logs cleared"
    }
